libraries/aws/provisioning/src/aws_iot_provisioning_parser.c

```diff
@@ -178,111 +197,90 @@ static AwsIotProvisioningError_t _parseRejectedResponse( IotSerializerDecoderObj
+    AwsIotProvisioning_Assert( pOperationString != NULL );
+
     AwsIotProvisioningError_t status = AWS_IOT_PROVISIONING_SUCCESS;
     IotSerializerError_t serializerStatus = IOT_SERIALIZER_SUCCESS;
 
-    AwsIotProvisioning_Assert( payloadDecoder != NULL );
-    AwsIotProvisioning_Assert( certPemDecoder != NULL );
-    AwsIotProvisioning_Assert( certIdDecoder != NULL );
-    AwsIotProvisioning_Assert( ownershipTokenDecoder != NULL );
-
-    /* Look for the certificate PEM data. */
+    /* Look for the data entry in the map. */
     serializerStatus = _pAwsIotProvisioningDecoder->find( payloadDecoder,
-                                                          PROVISIONING_CREATE_KEYS_AND_CERTIFICATE_RESPONSE_PAYLOAD_CERTIFICATE_PEM_STRING,
-                                                          certPemDecoder );
+                                                          pKeyString,
+                                                          payloadEntryDecoder );
 
     if( serializerStatus == IOT_SERIALIZER_NOT_FOUND )
     {
-        /* Cannot find "certificatePem" */
+        /* Entry not found in the payload map. */
         IotLogError( "parser: Unable to parse server response: Key entry not found in response:"
                      "ExpectedKey={\"%s\"}: Operation={%s}",
-                     PROVISIONING_CREATE_KEYS_AND_CERTIFICATE_RESPONSE_PAYLOAD_CERTIFICATE_PEM_STRING,
-                     CREATE_KEYS_AND_CERTIFICATE_OPERATION_LOG );
+                     pKeyString,
+                     pOperationString );
         status = AWS_IOT_PROVISIONING_BAD_RESPONSE;
     }
     else if( serializerStatus != IOT_SERIALIZER_SUCCESS )
     {
         status = AWS_IOT_PROVISIONING_INTERNAL_FAILURE;
     }
-    else if( certPemDecoder->type != IOT_SERIALIZER_SCALAR_TEXT_STRING )
+    else if( payloadEntryDecoder->type != IOT_SERIALIZER_SCALAR_TEXT_STRING )
     {
         IotLogError(
             "parser: Unable to parse server response: Invalid value type of key entry in payload: "
             "Expected type is text string: Key={\"%s\"}, Operation={%s}",
-            PROVISIONING_CREATE_KEYS_AND_CERTIFICATE_RESPONSE_PAYLOAD_CERTIFICATE_PEM_STRING,
-            CREATE_KEYS_AND_CERTIFICATE_OPERATION_LOG );
+            pKeyString,
+            pOperationString );
         status = AWS_IOT_PROVISIONING_BAD_RESPONSE;
     }
     else
     {
         /* Nothing to do. */
     }
 
+    return status;
+}
+
+/*------------------------------------------------------------------*/
+
+static AwsIotProvisioningError_t _parseCommonCertInfoInResponse( IotSerializerDecoderObject_t * payloadDecoder,
+                                                                 IotSerializerDecoderObject_t * certPemDecoder,
+                                                                 IotSerializerDecoderObject_t * certIdDecoder,
+                                                                 IotSerializerDecoderObject_t * ownershipTokenDecoder )
```

Comment:
Shouldn't we prefix all pointers with `p`, as per our [style guide](https://docs.aws.amazon.com/freertos/latest/lib-ref/c-sdk/main/guide_developer_styleguide.html#guide_developer_styleguide_naming_variables)?

Reply:
Not a fan of the Hungarian-esque notation of variable naming for pointer...But I will Disagree & Commit.